client/deposits/declarations.d.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * External dependencies
+ */
+
+/**
+ * Internal dependencies
+ */
+declare module '@woocommerce/components' {
+	const SummaryListPlaceholder: ( props: {
+		numberOfItems: number;
+	} ) => JSX.Element;
+
+	const SummaryList: ( props: {
+		label: string;
+		children: any; // TODO: figure out this and 2 lines below.
+	} ) => JSX.Element;
+
+	const OrderStatus: ( props: {
+		className?: string;
+		// eslint-disable-next-line @typescript-eslint/ban-types
+		orderStatusMap: object;
+		// eslint-disable-next-line @typescript-eslint/ban-types
+		order: object;
+	} ) => JSX.Element;
```

Comment:
I'm having problems figuring out these types. For example in the `@woocommerce/component` node_modules we have this 

```OrderStatus.propTypes = {
	/**
	 * The order to display a status for.
	 */
	order: PropTypes.object.isRequired,
	/**
	 * Additional CSS classes.
	 */
	className: PropTypes.string,
	/**
	 * A map of status to label for order statuses.
	 */
	orderStatusMap: PropTypes.object,
};
```
but I don't understand the difference between `PropTypes.object` and `PropTypes.object.isRequired`.